core/config/yaml_test.go

```diff
@@ -146,3 +146,26 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
 	empty := reflect.New(reflect.TypeOf(es)).Elem().Interface()
 	assert.True(t, reflect.DeepEqual(empty, es))
 }
+
+type pointerStruct struct {
+	MyBool *bool `yaml:"myBool"`
+}
+
+var myBoolyaml = []byte(`
+pointerStruct:
+  myBool: true
+`)
+
+func TestBoolPointer(t *testing.T) {
+	provider := NewProviderGroup("global", NewYAMLProviderFromBytes(myBoolyaml))
+	ps := pointerStruct{}
+	provider.GetValue("pointerStruct").PopulateStruct(&ps)
+	assert.True(t, *ps.MyBool)
+}
```

Comment:
for completeness sake, can we also check an explicit `false`? it'd be a shame if we messed that up